src/services/fungibleTokens.js

```diff
@@ -0,0 +1,89 @@
+import BN from 'bn.js';
+import * as nearApiJs from 'near-api-js';
+
+import sendJson from '../tmp_fetch_send_json';
+import { parseTokenAmount } from '../utils/amounts';
+import { ACCOUNT_HELPER_URL } from '../utils/wallet';
+
+const {
+    transactions: {
+        functionCall
+    },
+    utils: {
+        format: { 
+            parseNearAmount
+        }
+    }
+} = nearApiJs;
+
+const FT_MINIMUM_STORAGE_BALANCE = parseNearAmount('0.00125');
+const FT_STORAGE_DEPOSIT_GAS = parseNearAmount('0.00000000003');
+const FT_TRANSFER_GAS = parseNearAmount('0.00000000003');
+const FT_TRANSFER_DEPOSIT = '1'; // 1 yoctoNear
+
+export class FungibleTokens {
+    constructor(account) {
+        this.account = account;
+    }
+
+    async isStorageBalanceAvailable(contractName, accountId) {
+        return new BN((await this.getStorageBalance(contractName, accountId)).total).gte(new BN(FT_MINIMUM_STORAGE_BALANCE));
+    }
+
+    async getStorageBalance(contractName, accountId) {
+        return await this.account.viewFunction(contractName, 'storage_balance_of', { account_id: accountId });
+    }
+
+    async transfer(contractName, amount, memo, receiver, contractMetadata) {
```

Comment:
Is `contractName` already in `contractMetadata`?  If so, it would be preferable to use `contractMetadata.contractName` rather than duplicate this information here.

Since we don't actually have a use-case right now where we want to look up the metadata as part of calling `transfer()` (becausee we always want to use the metadata we've already loaded), and we are the only dependency on this code, let's not add a call to the block chain to this function chain _until and **if** we need it_; code that is not being used is a liability that doesn't provide any value -- we want code that exists here to exist _because it is useful_  and is _being used_.  Also, per my prior comment, if we _do_ want to provide loading capabilities to this module, we'll want to model that as a caching provider that can be passed in or constructed when an instance of this object is created; we _really_ don't want to encourage this method reaching out to hit the `viewFunction` _every time it's called_.  _In general_, we don't want extremely low level methods being responsible for _directly_ fetching data in cases where we know the data can and should be cached :). If methods do need to be responsible for fetching data that we expect should be cached, we'll want to model it using something like `dataloader` instances that can be passed to the object -- but for now, we don't need to solve that problem since we will have a source of truth (redux) for contract metadata that we can use to pass it into these calls _directly_, making this object much simpler.

With the elimination of this chain being responsible for loading metadata, we don't need `calculateAmount()` at all, either; we can just use `parseTokenAmount` directly on the metaData passed in to `transfer()`

Also, please convert this function to use an object as parameters per my prior comment, for the reasons I outlined there

...e.g.
```javascript
async transfer({ contractMetadata: { contractName, decimal }, amount, memo, receiver }) {...}
```

Reply:
> Is contractName already in contractMetadata?

No, I mean `ft_metadata` doesn't returns `contractName`, but I'm adding it in `getMetadata`. 

> With the elimination of this chain being responsible for loading metadata, we don't need calculateAmount() at all

Right, already removed it before reading this comment :) 

Done.